src/main/java/uk/ac/cam/cl/dtg/util/locations/PostCodeIOLocationResolver.java

```diff
@@ -185,40 +186,69 @@ private List<PostCode> carryOutExternalPostCodeServiceRequest(final List<String>
     private List<PostCode> submitPostCodeRequest(final List<String> unknownPostCodes)
             throws LocationServerException {
 
-        if (unknownPostCodes.size() > POSTCODEIO_MAX_REQUESTS) {
+        List<String> outCodes = Lists.newArrayList();
+        List<String> completePostCodes = Lists.newArrayList();
+
+        // Just filter by length, the regex for this would be too complex
+        for (String postCode : unknownPostCodes) {
+            if (postCode.length() > 4) {
+                completePostCodes.add(postCode);
+            }
+            else {
+                outCodes.add(postCode);
+            }
+        }
+
+        if (completePostCodes.size() > POSTCODEIO_MAX_REQUESTS) {
```

Comment:
We should certainly be checking the number of outCodes against this too.
```suggestion
        if (completePostCodes.size() + outCodes.size() > POSTCODEIO_MAX_REQUESTS) {
```

Reply:
Given that `POSTCODEIO_MAX_REQUESTS` is based on the max. number of postcodes that can be batch-requested from the `/postcodes` endpoint in one request, comparing the combined number of postcodes and outcodes to this limit feels fairly arbitrary - but there should definitely be a limit, and this works as well as any.

Reply:
Ah, okay. From a user perspective this keeps things consistent with how they were before which I prefer, though admittedly I struggle to see anyone using 100 different postcodes in one request anyway.